hw4.js

```diff
@@ -0,0 +1,475 @@
+    return result;
+};
+
+/**
+ * Exercise 2
+ *
+ * Write a function that returns the smallest value of an array
+ * [4,2,10,27] => 2
+ */
+let arr2 = [4, 2, 10, 27];
+const getSmallestValue = (numbers) => {
+    let result = 0;
+    for (let i = 0; i < arr2.length; i++) {
+        if (arr2[0] > arr2[i]) {
+            result = arr2[i];
+        }
+    }
+    return result;
+};
+
+/**
+ * Exercise 3
+ *
+ * Write a function that returns the biggest value of an array
+ * [5,22,9,43] => 43
+ */
+let arr3 = [5, 22, 9, 43];
+const getBiggestValue = (numbers) => {
+    let result = 0;
+    for (let i = 0; i < arr3.length; i++) {
+        if (arr3[0] < arr3[i]) {
+            result = arr3[i];
+        }
+    }
+    return result;
+};
+
+/**
+ * Exercise 4
+ *
+ * Write a function that takes an array of strings as input
+ * and returns only those shorter than 20 characters
+ *
+ *[
+ * 'I am a short string',
+ * 'I seem to be short too',
+ * 'And I am a long string'
+ *] => [
+ * 'I am a short string',
+ * 'And I am a long string' //looks like an error in the task here
+ *]
+ *
+ * Use: filter
+ */
+let arr4 = ["I am a short string", "I seem to be short too", "And I am a long string"];
+const getShorterStrings = (strings, characters = 20) => {
+    const result = arr4.filter((string) => string.length < 20);
+    return result;
+};
+
+/**
+ * Exercise 5
+ *
+ * Write a function that takes the following data as input:
+ *
+ *[
+ * { name: 'shark', likes: 'ocean' },
+ * { name: 'turtle', likes: 'pond' },
+ * { name: 'otter', likes: 'fish biscuits' }
+ *]
+ *
+ * And returns an array of strings:
+ *
+ * [ 'shark likes ocean', 'turtle likes pond', 'otter likes fish biscuits' ]
+ *
+ * Use: map
+ */
+let arr5 = [
+    { name: "shark", likes: "ocean" },
+    { name: "turtle", likes: "pond" },
+    { name: "otter", likes: "fish biscuits" },
+];
+const getComputedStrings = (fish) => {
+    const result = arr5.map((el) => {
+        return `${el.name} likes ${el.likes}`;
+    });
+    return result;
+};
+
+/**
+ * Exercise 6
+ *
+ * Write a function that takes 2 objects as input and returns 1 with
+ * common properties. If properties have the same keys use the latter.
+ *
+ * [{ name: 'Alice' }, { age: 11 }] => { name: 'Alice', age: 11 }
+ *
+ * We use: ...
+ */
+let obj1 = { name: "Alice" };
+let obj2 = { age: 11 };
+const mergeObjects = (objects) => {
+    const result = { ...obj1, ...obj2 };
+    return result;
+};
+
+/**
+ * Exercise 7
+ *
+ * Write a function that returns the smallest value of an array
+ * [5,200,-5,41] => -5
+ *
+ * Use: operator ... and Math.min
+ */
+let arr7 = [5, 200, -5, 41];
+const getSmallestValue2 = (numbers) => {
+    let result = Math.min(...arr7);
+    return result;
+};
+
+/**
+ * Exercise 8
+ *
+ * Write a function that returns odd array values.
+ * [77,2,30,51] => [77,51]
+ *
+ * Use: reduce
+ */
+let arr8 = [77, 2, 30, 51];
+const getOddValues2 = (numbers) => {
+    let reducer = (a, b) => (b % 2 === 0 ? a : a.concat(b));
```

Comment:
concat is used for concat array, but here you need to add item into the array, so there you can use a.**push**(b)